routing/routing_integration_tests/bicycle_route_test.cpp

```diff
@@ -78,99 +96,119 @@ UNIT_TEST(NetherlandsAmsterdamSingelStCyclewayOpposite)
 UNIT_TEST(RussiaMoscowKashirskoe16ToCapLongRoute)
 {
   // There is no dedicated bicycle infra, except short end part. All OSM routers give different results,
-  // but OM and Valhalla looks better.
+  // OM yields a short and logical route shortcutting via some service roads and footways (allowed in Russia).
   CalculateRouteAndTestRouteLength(
       GetVehicleComponents(VehicleType::Bicycle),
       mercator::FromLatLon(55.66230, 37.63214), {0., 0.},
-      mercator::FromLatLon(55.68927, 37.70356), 7323.01);
+      mercator::FromLatLon(55.68927, 37.70356), 6968.64);
 }
 
-// Passing through living_street and service are allowed in Russia
-UNIT_TEST(RussiaMoscowServicePassThrough1)
+UNIT_TEST(Germany_UseServiceCountrysideRoads)
 {
-  TRouteResult route =
-        integration::CalculateRoute(integration::GetVehicleComponents(VehicleType::Bicycle),
-                                    mercator::FromLatLon(55.66230, 37.63214), {0., 0.},
-                                    mercator::FromLatLon(55.68895, 37.70286));
-  TEST_EQUAL(route.second, RouterResultCode::NoError, ());
+  // Goes by smaller roads, including service ones. Also avoids extra 60m uphill
+  // of the secondary road route. Most similar to Valhalla, but 2km shorter.
+  // https://github.com/organicmaps/organicmaps/issues/6027
+  CalculateRouteAndTestRouteLength(
+      GetVehicleComponents(VehicleType::Bicycle),
+      mercator::FromLatLon(49.1423, 10.068), {0., 0.},
+      mercator::FromLatLon(49.3023, 10.5738), 47769.2);
 }
 
-UNIT_TEST(RussiaMoscowServicePassThrough2)
+UNIT_TEST(RussiaMoscowServicePassThrough)
 {
+  // Passing through living_street and service is allowed in Russia.
```

Comment:
this is AFAIK typically allowed also elsewhere? At least on service which is not `driveway`

Reply:
@vng should know the context better here